lib/common/util/exceptions.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import moment from 'moment'
+
+import type {ParsedRangesAndDates} from '../../types'
+
+// Method to filter out new dates (0's) so that flow is happy
+// We have to use a reducer to keep flow happy.
+export function filterOutNewDates (dates: Array<string | 0>): Array<string> {
+  const outputDates: Array<string> = []
+  const filter = (acc, current) => {
+    typeof current === 'string' && acc.push(current)
+    return acc
+  }
+  dates.reduce(filter, outputDates)
+  return outputDates
+}
+
+/**
+ * This method adds a range of dates from startRange to endRange to the dates array, inclusively according to startInclusive
+ * and endInclusive.
+ */
+export function addRangeOfDates (
```

Comment:
I would refactor this method to be a recursive method which only accepts `dates` and `endRange`. The method would return `dates` if `endRange` is in `dates`.  I'd then also create a second method to do the same but with `startRange`. These methods can then be combined to re-create the current functionality with less complexity

Reply:
You bet this is recursive now in https://github.com/ibi-group/datatools-ui/pull/828/commits/0bd970887b85994ed9f9452ec5640b01eb021f28